python/deltalake/writer.py

```diff
@@ -152,6 +176,11 @@ def visitor(written_file: Any) -> None:
         schema=schema if not isinstance(data, RecordBatchReader) else None,
         file_visitor=visitor,
         existing_data_behavior="overwrite_or_ignore",
+        file_options=file_options,
+        max_open_files=max_open_files or 1024,
+        max_rows_per_file=max_rows_per_file or 0,
+        min_rows_per_group=min_rows_per_group or 0,
+        max_rows_per_group=max_rows_per_group or 1024 * 1024,  # pyarrow default
```

Comment:
curious why not set the default values through method argument defaults?

Reply:
yeah good point, will update